lib/src/rules/avoid_leading_underscore_for_non_private_identifiers.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2015, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/utils.dart';
+
+const _desc = r'Avoid leading underscore for identifiers that aren’t private.';
+
+const _details = r'''
+
+**DON’T** use a leading underscore for identifiers that aren’t private.
+
+Dart uses a leading underscore in an identifier to mark members and top-level declarations as private. This trains users to associate a leading underscore with one of those kinds of declarations. They see “_” and think “private”.
```

Comment:
nit: these lines should be wrapped to fit within 80 columns.

Reply:
Wrapped lines to fit within 80 columns.